test/Microsoft.ML.AutoML.Tests/AutoFitTests.cs

```diff
@@ -320,6 +322,36 @@ public void AutoFitWithPresplittedData()
 
         }
 
+        [Fact]
+        public void AutoFitMaxExperimentTimeTest()
+		{
+            // 1 Binary classification experiment takes less than 5 seconds.
+            // System.OperationCanceledException is thrown when ongoing experiment
+            // is canceled and at least one model has been generated.
+            var context = new MLContext(1);
+            var dataPath = DatasetUtil.GetUciAdultDataset();
+            var columnInference = context.Auto().InferColumns(dataPath, DatasetUtil.UciAdultLabel);
+            var textLoader = context.Data.CreateTextLoader(columnInference.TextLoaderOptions);
+            var trainData = textLoader.Load(dataPath);
+            var experiment = context.Auto()
+                .CreateBinaryClassificationExperiment(5)
+                .Execute(trainData, new ColumnInformation() { LabelColumnName = DatasetUtil.UciAdultLabel });
+            RunDetail<BinaryClassificationMetrics>[] runDetails = experiment.RunDetails.Where(r => r.Model != null).ToArray();
+            foreach(RunDetail<BinaryClassificationMetrics> runDetail in runDetails)
+			{
+                ModelContainer modelContainer = GetInstanceField(typeof(RunDetail<BinaryClassificationMetrics>), runDetail, "_modelContainer") as ModelContainer;
+                MLContext thisContext = GetInstanceField(typeof(ModelContainer), modelContainer, "_mlContext") as MLContext;
+                Assert.True((thisContext.Model.GetEnvironment() as ICancelable).IsCanceled);
+            }
+        }
```

Comment:
I might recommend extending this unit test to test the ability to refit/transform/evaluate the BestRun after canceling: _(or perhaps as its own test)_
```suggestion
            var refitModel = experiment.BestRun.Estimator.Fit(trainData);
            IDataView predictions = refitModel.Transform(trainData);
            var metrics = mlContext.BinaryClassification.Evaluate(predictions, labelColumnName: DatasetUtil.UciAdultLabel);
            assert(metrics?.Accuracy > 0.5); 
        }
```

Reply:
Added. I've also removed using reflection to check if the last model is canceled, as the `RunDetails` of the last model has an `Exception` field which contains the "Operation was canceled" message of the `OperationCanceledException`. :)

Reply:
Good find.